packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -31,34 +34,43 @@ abstract contract FundingRateApplier is Testable, Lockable {
      * FUNDING RATE APPLIER DATA STRUCTURES *
      ****************************************/
 
-    // Locally stores the finder.
-    // This is private because it is expected that another contract, like the FeePayer, will publicly expose the finder
-    // address.
-    FinderInterface private finder;
-
-    // Last time the `cumulativeFundingRateMultiplier` was updated.
-    uint256 public lastUpdateTime;
+    struct FundingRate {
+        // Current funding rate value.
+        FixedPoint.Signed rate;
+        // Identifier to retrieve the funding rate.
+        bytes32 identifier;
+        // Per second reward paid to get funding rate proposers.
+        FixedPoint.Unsigned rewardRate;
+        // Bond percentage.
+        FixedPoint.Unsigned bondPercentage;
+        // Tracks the cumulative funding payments that have been paid to the sponsors.
+        // The multiplier starts at 1, and is updated by computing cumulativeFundingRateMultiplier * (1 + effectivePayment).
+        // Put another way, the cumulativeFeeMultiplier is (1 + effectivePayment1) * (1 + effectivePayment2) ...
+        // For example:
+        // The cumulativeFundingRateMultiplier should start at 1.
+        // If a 1% funding payment is paid to sponsors, the multiplier should update to 1.01.
+        // If another 1% fee is charged, the multiplier should be 1.01^2 (1.0201).
+        FixedPoint.Unsigned cumulativeMultiplier;
+        // Most recent time that the funding rate was updated.
+        uint256 updateTime;
+        // Most recent time that the funding rate was applied and changed the cumulative multiplier.
+        uint256 applicationTime;
+        // The time for the active (if it exists) funding rate proposal. 0 otherwise.
+        uint256 proposalTime;
+    }
 
-    // Timestamp used in case of emergency shutdown.
-    uint256 public emergencyShutdownTimestamp;
+    FundingRate public fundingRate;
 
-    // Tracks the cumulative funding payments that have been paid to the sponsors.
-    // The multiplier starts at 1, and is updated by computing cumulativeFundingRateMultiplier * (1 + effectivePayment).
-    // Put another way, the cumulativeFeeMultiplier is (1 + effectivePayment1) * (1 + effectivePayment2) ...
-    // For example:
-    // The cumulativeFundingRateMultiplier should start at 1.
-    // If a 1% funding payment is paid to sponsors, the multiplier should update to 1.01.
-    // If another 1% fee is charged, the multiplier should be 1.01^2 (1.0201).
-    FixedPoint.Unsigned public cumulativeFundingRateMultiplier;
+    uint256 emergencyShutdownTimestamp;
```

Comment:
did you intend to leave out the `public` key word?

Reply:
Nope, nice catch -- this was a typo when I was resolving merge conflicts.